### PR TITLE
Add TaskValidation module

### PR DESCRIPTION
**Pull request by ratsclub**
## Proposed Changes

This is a PR for: #228. It adds the `TaskValidation` module.

## Types of changes

What types of changes does your code introduce to FsToolkit.ErrorHandling?
_Put an `x` in the boxes that apply and remove ones that don't apply_ 

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] Build and tests pass locally
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/FsToolkit.ErrorHandling.TaskResult/TaskValidation.fs [hunk 1]
```diff
@@ -0,0 +1,193 @@
+﻿namespace FsToolkit.ErrorHandling
+
+open System.Threading.Tasks
+
+/// TaskValidation<'a, 'err> is defined as Async<Result<'a, 'err list>> meaning you can use many of the functions found in the Result and Async module.
```
Comment (TheAngryByrd) [inline src/FsToolkit.ErrorHandling.TaskResult/TaskValidation.fs:5]:
```suggestion
/// TaskValidation<'a, 'err> is defined as Task<Result<'a, 'err list>> meaning you can use many of the functions found in the Result and Async module.
```

File: tests/FsToolkit.ErrorHandling.TaskResult.Tests/TaskValidation.fs [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+
+#if FABLE_COMPILER_PYTHON
+open Fable.Pyxpecto
+#endif
+#if FABLE_COMPILER_JAVASCRIPT
+open Fable.Mocha
+#endif
+#if !FABLE_COMPILER
+open Expecto
+#endif
```
Comment (TheAngryByrd) [inline tests/FsToolkit.ErrorHandling.TaskResult.Tests/TaskValidation.fs:11]: I don't think we need the Fable ifdefs for the Task test suite.

File: src/FsToolkit.ErrorHandling.TaskResult/TaskValidationCE.fs [hunk 3]
```diff
@@ -0,0 +1,191 @@
+﻿namespace FsToolkit.ErrorHandling
+
+open System
+open System.Threading.Tasks
+
+[<AutoOpen>]
+module TaskValidationCE =
+
+    type TaskValidationBuilder() =
```
Comment (TheAngryByrd) [inline src/FsToolkit.ErrorHandling.TaskResult/TaskValidationCE.fs:9]: Seems to be missing the [ResumableStateMachine](https://github.com/fsharp/fslang-design/blob/main/FSharp-6.0/FS-1087-resumable-code.md). See how `TaskResult` is implemented.

**Review by TheAngryByrd — CHANGES_REQUESTED** [review]
Thanks for this! Left some comments on stuff that needs fixed.

File: src/FsToolkit.ErrorHandling.TaskResult/TaskValidationCE.fs (hunk 3 above)
Comment (ratsclub) [reply src/FsToolkit.ErrorHandling.TaskResult/TaskValidationCE.fs:9]: I have never dealt with this before, I'll read about it and try to implement it.

**Comment by TheAngryByrd** [comment]
👋 I'm gonna close this since there hasn't been movement in a year. Also TaskResults have moved to the core library if you feel motivated to give this another go.

**Pull request closed without merging**